synapse/handlers/event_auth.py

```diff
@@ -102,17 +107,17 @@ async def has_restricted_join_rules(
         join_rules_event = await self._store.get_event(join_rules_event_id)
         return join_rules_event.content.get("join_rule") == JoinRules.MSC3083_RESTRICTED
 
-    async def get_spaces_that_allow_join(
+    async def get_rooms_that_allow_join(
         self, state_ids: StateMap[str]
     ) -> Collection[str]:
         """
-        Generate a list of spaces which allow access to a room.
+        Generate a list of rooms which membership in allow access to a room.
```

Comment:
I think this is a marginal improvement, ymmv.

```suggestion
        Generate a list of rooms in which membership allows access to a room.
```